### PR TITLE
Improve HTTPS detection accuracy in Site Health

### PR DESCRIPTION
**Pull request by felixarntz**
This PR addresses follow-up feedback to https://core.trac.wordpress.org/changeset/49904 from https://github.com/WordPress/wordpress-develop/pull/870#issuecomment-763015259 (that PR is related, but targets another ticket).

It also renames a function as suggested by [this comment](https://core.trac.wordpress.org/ticket/47577#comment:31).

Trac ticket: https://core.trac.wordpress.org/ticket/47577

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-admin/includes/class-wp-site-health.php [hunk 1]
```diff
@@ -1498,6 +1498,10 @@ public function get_test_is_in_debug_mode() {
 	 * @return array The test results.
 	 */
 	public function get_test_https_status() {
+		// Enforce fresh HTTPS detection results. This is normally invoked by using cron, but for Site Health it should
+		// always rely on the latest results.
+		wp_update_https_detection_errors();
```
Comment (adamsilverstein) [inline src/wp-admin/includes/class-wp-site-health.php:1503]: Great!

File: src/wp-includes/https-detection.php [hunk 2]
```diff
@@ -9,28 +9,53 @@
...
 	// current request is using.
 	/** This filter is documented in wp-includes/link-template.php */
 	$site_url = apply_filters( 'site_url', get_option( 'siteurl' ), '', null, null );
 
-	if ( 'https' !== wp_parse_url( $site_url, PHP_URL_SCHEME ) ) {
-		return false;
-	}
-
-	return true;
+	return 'https' === wp_parse_url( $site_url, PHP_URL_SCHEME );
```
Comment (adamsilverstein) [inline src/wp-includes/https-detection.php:58]: nice

**Comment by adamsilverstein** [comment]
👍🏼 Overall code changes look great; i'll give this a test locally again as well.

File: src/wp-admin/includes/class-wp-site-health.php (hunk 1 above)
Comment (TimothyBJacobs) [reply src/wp-admin/includes/class-wp-site-health.php:1503]: Should this be an async test now so we don't slow down page load?

**Comment by TimothyBJacobs** [comment]
How do you feel about making `wp_update_https_detection_errors` shortcircuitable? The reason I ask is I maintain a plugin that functions similarly, but uses an external server to ping the site to check for HTTPS to account for instances where a loopback request results in a false negative. Ideally, we'd be able to override the check to use the more accurate detection, but still keep the user in Site Health.

**Comment by felixarntz** [comment]
@TimothyBJacobs Great suggestion - I'll add that tomorrow.

**Comment by adamsilverstein** [comment]
@felixarntz I did some testing locally and took some screenshots. Noticed a couple of tiny issues:

Accessing wp-admin via http (with my invalid "self signed" SSL cert), I noticed an issue where the title is getting some extraneous encoding:

image

When I changed my WordPress Address to **http** I got a different message, looks good:

image

Site address only set to **http**, looks good:
image

Both set to **http**, switched to accessing wp-admin via http:

There is a slight problem here, the space is missing after the work "There":

image

**Comment by felixarntz** [comment]
@adamsilverstein @TimothyBJacobs Addressed your feedback.

Here are some screenshots of the different states.

**HTTPS not set up, not supported**
Screen Shot 2021-01-28 at 12 19 39 PM

**HTTPS not set up, but supported**
Screen Shot 2021-01-28 at 12 16 37 PM

**HTTPS not set up, but supported, with "WordPress Address" already using HTTPS though**
Screen Shot 2021-01-28 at 12 17 55 PM

**HTTPS set up, but not supported (i.e. there are problems)**
Screen Shot 2021-01-28 at 12 15 33 PM

@TimothyBJacobs One question you might know the answer to: Now that I've made the test asynchronous and added a REST endpoint for it, do I need to update `wp-api-generated.js`? At what point is that generated and how?

**Comment by felixarntz** [comment]
Committed via https://core.trac.wordpress.org/changeset/50072.

**Pull request closed without merging**